src/protocol.rs

```diff
@@ -85,14 +56,32 @@ pub enum Event<Response> {
     },
     #[serde(rename = "complete")]
     Complete { id: String },
+    #[serde(rename = "connection_ack")]
+    ConnectionAck { payload: Option<serde_json::Value> },
```

Comment:
Nice.  Not entirely sure why I had a whole struct & `Serialize` impl for this before.  Silly me

Reply:
Yeah :) A minor downside is that the `id` is now an option and we have to add match arms in the main receiver loop for events that should never occur there.